### DISCUSSION TITLE
How to save form fields?

### DISCUSSION THREAD

**Discussion by joshishaswat** [aannirajpatel/samosabucket#4] [Q&A] [answered]
We added two more form fields for the user profile and were able to configure the JSON schema to send any submissions in the new fields to our MongoDB. However, after refreshing the page:

<img width="637" alt="Screen Shot 2021-05-14 at 12 09 36 PM" src="https://user-images.githubusercontent.com/50994221/118298348-5bf42180-b4ad-11eb-8323-ddecb15fc29f.png">

The new form fields reset to a blank state on the frontend although they are still stored in our MongoDB instance when we wanted the "venmo" and "allergies" tabs to be saved just like the other fields. 

We made the following adjustments:

**1) Under ../frontend/src/Me.js**

```
  const [venmo, setVenmo] = useState("");
```

```
  const venmoChange = (e) => {
    setVenmo(e.target.value);
  };
```

`      venmo: venmo,` added in the updateSubmit arrow function

**2) Updated our fields in User.js for the mongoose schema (this was structured fine in our database)**

**3)  Under ../routes/user.js**

We added our venmo field to the async function

`    const { email, password, name, address, phone, venmo, other } = req.body;
`

And user variable 

`      user = new User({ email, password, name, address, phone, venmo, other });
`

Have we missed a step when configuring the middleware?

**Comment by aannirajpatel** [comment, marked as answer]
useState("") seems to be the issue. You need to write code that fetches and sets those fields every time the user reloads - if I remember correctly the "Me" component is coded up so as to fetch user data and show it whenever you reload.

Edit: by useState("") being the issue, I mean that you can keep useState("") line as-is, but then when the component loads, you need to write code somewhere that does the fetching and sets the state.

**Reply by joshishaswat** [reply]
Perfect updating our useState("") resolved our updating problem
